### PR TITLE
Ruby: Do not expose AST layer through `ruby.qll`

### PR DESCRIPTION
**Pull request by hvitved**
Since writing data-flow queries is much more common than writing syntactic queries, it makes little sense to have the AST layer exposed by default in `ruby.qll`.

This PR therefore removes that AST libraries from `ruby.qll` (instead, one should import `codeql.ruby.AST`). In fact, `ruby.qll` is now completely empty, and is only kept for backwards compatibility with quick queries.

### CODE REVIEW

**Comment by aibaars** [comment]
Shall we remove all `import ruby` statements, and replace them with `import codeql.Locations` wherever we actually need that import?

**Comment by hvitved** [comment]
> Shall we remove all `import ruby` statements, and replace them with `(private) import codeql.Locations` wherever we actually need that import?

Or perhaps we should just let `codeql.ruby.AST` reexport `codeql.Locations`?

**Comment by aibaars** [comment]
> > Shall we remove all `import ruby` statements, and replace them with `(private) import codeql.Locations` wherever we actually need that import?
> 
> Or perhaps we should just let `codeql.ruby.AST` reexport `codeql.Locations`?

Fine with me too, the most important thing to me is removing all `import ruby` statements. I'm not really convinced that the AST module should expose `Locations` though. It's a bit arbitrary, why would AST do that, but CFG or dataflow not.

**Comment by hvitved** [comment]
> It's a bit arbitrary, why would AST do that, but CFG or dataflow not.

I think I would let those libraries expose locations as well, since it is such a core concept.

**Comment by aibaars** [comment]
> I think I would let those libraries expose locations as well, since it is such a core concept.

Having all of them export locations is good; definitely better than having one (or some of them) do it.

File: ruby/ql/lib/codeql/ruby/AST.qll [hunk 1]
```diff
@@ -17,6 +17,7 @@ private import ast.internal.AST
 private import ast.internal.Scope
 private import ast.internal.Synthesis
 private import ast.internal.TreeSitter
+private import Customizations
```
[the remark this answers is not in the document]
Comment (hvitved) [reply ruby/ql/lib/codeql/ruby/AST.qll:20]: Is was done in `ruby.qll` previously, and we need to make sure that it is always in scope. Adding it here should effectively achieve that, as the CFG layer depends on the AST layer, and the data-flow layer depends on the CFG layer.

[the remark this answers is not in the document]
Comment (aibaars) [reply ruby/ql/lib/codeql/ruby/AST.qll:20]: Makes sense, thanks for clarifying.

**Comment by asgerf** [comment]
Wouldn't it make sense to make the three layers available as named modules?

So after `import ruby` you can use the prefixes `AST::`, `CFG::` and `DataFlow::`.

**Comment by hvitved** [comment]
> Wouldn't it make sense to make the three layers available as named modules?
> 
> So after `import ruby` you can use the prefixes `AST::`, `CFG::` and `DataFlow::`.

Fine by me.

File: ruby/ql/lib/change-notes/2022-09-14-ruby-qll.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
 ---
 category: breaking
 ---
-* `import ruby` no longer brings the standard Ruby AST libarary into scope (it brings nothing into scope, so should no longer be used). Instead, import the AST library via `import codeql.ruby.AST`.
\ No newline at end of file
+* `import ruby` no longer brings the standard Ruby AST libarary into scope; it instead brings a module `Ast` into scope, which must be imported. Alternatively, it is also possible to import `codeql.ruby.AST`.
```
Comment (nickrolfe) [inline ruby/ql/lib/change-notes/2022-09-14-ruby-qll.md:4]:
```suggestion
* `import ruby` no longer brings the standard Ruby AST library into scope; it instead brings a module `Ast` into scope, which must be imported. Alternatively, it is also possible to import `codeql.ruby.AST`.
```

**Review by nickrolfe — APPROVED** [review]
LGTM, and we agreed in the team meeting to go with this approach, so feel free to merge.

As a follow-up, we could perhaps fix some of those QL-for-QL alerts by changing the casing on some of the existing imports that are `as AST` or `as CFG`.

**Pull request merged**